Sign up flow

drillprop opened the pull request:
Fix #4134

drillprop committed:
signup steps ground work

drillprop committed:
password step ground work

drillprop committed:
seed step ground work

drillprop committed:
email validation

drillprop committed:
dialog fixes

drillprop committed:
wip

drillprop committed:
create separate sign up modal

drillprop committed:
remove redundant styles

drillprop committed:
signup with membership wip

drillprop committed:
remove useRegister hook

drillprop committed:
persist data between steps

drillprop committed:
fix button at the end of the form

drillprop committed:
show confetti behind modal only on bigger screens

drillprop committed:
show dividers

drillprop committed:
fix misc issues

drillprop committed:
handle email already taken error

WRadoslaw commented on packages/atlas/src/components/_overlays/Modal/Modal.styles.ts:
```diff
@@ -33,7 +33,8 @@ export const ModalContent = styled.div<ModalContentProps>`
   max-height: calc(100vh - var(--size-topbar-height));
-  overflow: hidden;
+
+  /* overflow: hidden; */
```
Comment

WRadoslaw reviewed (changes requested):
1. Should we show password on input focus when it's hidden?

![image](https://github.com/Joystream/atlas/assets/51168865/1e40008d-e548-4310-a37b-ea4c47f47292)

2. In designs, password screen don't have scrollbar and divider

![image](https://github.com/Joystream/atlas/assets/51168865/09ea8527-a049-4c2e-beef-4f6dfdcdaefb)


3. Not sure why I cannot create a membership (request to the faucet is not triggered)

![image](https://github.com/Joystream/atlas/assets/51168865/7796e534-9f01-4175-8a05-aa4c687129ae)

drillprop commented:
> 2. In designs, password screen don't have scrollbar and divider

I don't think that's true. https://www.figma.com/file/ymHYOFItrmFaLeBC5Z4tpQ/Sign-up-%2F-Log-in?type=design&node-id=2939-77080&t=AcDsClQRTl3N7XH9-4

Check the instructions for RWD 
![image](https://github.com/Joystream/atlas/assets/51168865/da21950e-673e-4bd5-a38c-e265604df2ba)

WRadoslaw commented:
> > 2. In designs, password screen don't have scrollbar and divider
> 
> I don't think that's true. https://www.figma.com/file/ymHYOFItrmFaLeBC5Z4tpQ/Sign-up-%2F-Log-in?type=design&node-id=2939-77080&t=AcDsClQRTl3N7XH9-4
> 
> Check the instructions for RWD ![image](https://user-images.githubusercontent.com/51168865/241231029-da21950e-673e-4bd5-a38c-e265604df2ba.png)

If so I'd add some spacing on the bottom, rn it looks kind of awkward.

![image](https://github.com/Joystream/atlas/assets/92513933/ff67160e-97c0-4a64-848c-c9a3acb070e3)

drillprop committed:
Merge remote-tracking branch 'upstream/accounts' into 4134-sign-up-flow

drillprop committed:
fix login modal styling inconsistency

drillprop committed:
cr fixes and minor adjustments

drillprop commented:
> 3. Not sure why I cannot create a membership (request to the faucet is not triggered)

For some reason, it's happening only on production. Investigating...

drillprop committed:
fix cryptoWaitReady issue

drillprop committed:
fix lib typings, improve UX

drillprop commented:
> > 3. Not sure why I cannot create a membership (request to the faucet is not triggered)
> 
> For some reason, it's happening only on production. Investigating...
 
Fixed. I also introduced some adjustments in your Login Modal because I spotted some styling inconsistencies

drillprop committed:
more fixes
